### DISCUSSION TITLE
LAMMPS build - "recompile with -fPIC" error

### DISCUSSION THREAD

**Discussion by Mskater** [deepmodeling/deepmd-kit#298] [Q&A]
New problem for LAMMPS build: 
 
  I had errors which said recompile with -fPIC due to a PIE error.   After poking around a lot, I added the flag to the build command:  make mpi -j4 -no-pie.
It ended with:
 -L/usr/lib/python2.7/config-x86_64-linux-gnu -L/usr/lib -lpython2.7 -lpthread -ldl  -lutil -lm  -Xlinker -export-dynamic -Wl,-O1 -Wl,-Bsymbolic-functions   -o ../lmp_mpi
size ../lmp_mpi
A whole list of object files (*.o) are in the Obj_mpi directory (under the LAMMPS source directory ).
Is this an error?    Because I cannot find an lmp_mpi executable.  
What to do now? Thanks for any help.
Ubuntu 18.04

**Comment by njzjz** [comment]
`lmp_mpi` should be in the `src` directory, and "a whole list of object files (*.o) are in the Obj_mpi directory" is expected.

**Comment by Mskater** [comment]
There is no lmp_mpi in the source directory.  Yes I know the object files should be in Obj_mpi.
The errors now go like this for every object file (when I don't have -no-pie as a flag in the make mpi -j4), when I do, it doesn't even get this far :

/usr/bin/ld: fix_npt_cauchy.o: relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC    
 (And on and on, naming all object files)
   It's telling me to recompile all the object files differently somehow? 
How to get over this error?  Thanks!

**Comment by njzjz** [comment]
What version of LAMMPS do you use? I think it's an issue related to LAMMPS itself. DeePMD-kit plugin does nothing to `fix_npt_cauchy`.

**Comment by Mskater** [comment]
Yes, thanks, I know what you mean, it's about LAMMPS, but I was following the instructions on this website to build LAMMPS, (traditonal make), and something is missing.   I am installing the March 2020 stable version.   I don't know how to "recompile with -fPIC", and have read that it may mean something else, and so I'm stuck again.
